hazelcast/src/main/java/com/hazelcast/util/StringUtil.java

```diff
@@ -127,4 +130,17 @@ public static String lowerCaseInternal(String s) {
         }
         return s.toLowerCase(LOCALE_INTERNAL);
     }
+
+    /**
+     * Returns a String representation of the time.
+     *
+     * This method is not particularly efficient since it generates a ton of litter.
+     *
+     * @param timeMillis time in millis
+     * @return the String
+     */
+    public static String timeToString(long timeMillis) {
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
```

Comment:
It isn't only the litter of the DateFormat object, but also during the format method, waste is generated. For its current purposes it is fine :)